zerver/webhooks/zapier/view.py

```diff
@@ -14,10 +14,22 @@
 def api_zapier_webhook(request: HttpRequest, user_profile: UserProfile,
                        payload: Dict[str, Any]=REQ(argument_type='body')) -> HttpResponse:
     subject = payload.get('subject')
-    content = payload.get('content')
+    topic = payload.get('topic')
+
     if subject is None:
-        return json_error(_("Subject can't be empty"))
+        if topic is not None:
+            subject_or_topic = topic
+        else:
+            return json_error(_("Subject and topic can't both be empty"))
+    else:
+        if topic is None:
+            subject_or_topic = subject
+        else:
+            return json_error(_("Specify one of subject and topic, not both"))
+
+    content = payload.get('content')
     if content is None:
         return json_error(_("Content can't be empty"))
-    check_send_webhook_message(request, user_profile, subject, content)
+
+    check_send_webhook_message(request, user_profile, subject_or_topic, content)
```

Comment:
I would also try to remove reference to `subject` from the code. So the variable should be `topic`, and it should be set to payload.get('subject') if needed.